qa/rpc-tests/checkdatasig-activation.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+This test checks activation of OP_CHECKDATASIG
+"""
+
+from test_framework.test_framework import ComparisonTestFramework
+from test_framework.util import satoshi_round, assert_equal, assert_raises_rpc_error
+from test_framework.comptool import TestManager, TestInstance, RejectResult
+from test_framework.blocktools import *
+from test_framework.script import *
+
+# far into the future
+MAGNETIC_ANOMALY_START_TIME = 2000000000
+
+# Error due to invalid opcodes
+BAD_OPCODE_ERROR = b'mandatory-script-verify-flag-failed (Opcode missing or not understood)'
+RPC_BAD_OPCODE_ERROR = "16: " + \
+    BAD_OPCODE_ERROR.decode("utf-8")
+
+
+class PreviousSpendableOutput():
+
+    def __init__(self, tx=CTransaction(), n=-1):
+        self.tx = tx
+        self.n = n  # the output we're spending
+
+
+class CheckDataSigActivationTest(ComparisonTestFramework):
+
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+        self.extra_args = [['-whitelist=127.0.0.1',
+                            "-magneticanomalyactivationtime=%d" % MAGNETIC_ANOMALY_START_TIME,
+                            "-replayprotectionactivationtime=%d" % (2 * MAGNETIC_ANOMALY_START_TIME)]]
+
```

Comment:
you can get rid of the -replayprotectionactivationtime and change the -magenticanomalyactivation time to BU's version of the activation time.

Reply:
Hmm, this must be code from the midst of my changeset? Because it already is adapted accordingly at the HEAD commit of this PR branch.